test/Subscriber/unsusbcribe.test.ts

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const sinon = require('sinon');
+import { assert } from 'chai'
+import { createMockClient } from 'mock-apollo-client';
+
+import Subscriber from './../../src/Subscriber'
+import GraphClient from './../../src/GraphClient'
+import SpyUnsubscribe from './utils/SpyUnsubscribe'
+
+describe('Subscriber.unsubscribe()', () => {
+  let apolloClient;
+  let graphClient;
+  let subscriptionQry;
+  let subscriber;
+
+  beforeEach(() => {
+    apolloClient = createMockClient();
+    graphClient = new GraphClient(apolloClient);
+    subscriptionQry = 'subscription{stakeRequesteds{id}}';
+  });
+
+  it('should work with correct parameters', async () => {
+    sinon.replace(
+      graphClient,
+      'subscribe',
+      sinon.fake.resolves(new SpyUnsubscribe()),
```

Comment:
I think that we should spy on the actual `querySubscription` object, instead of using `SpyUnsubscribe`.

Reply:
For that Subscription object needs to be created. Here apollo exposes Subscription as type and not class. I think that class is encapsulated in the apollo client. Any reference for creating Subscription object will be helpful.

Reply:
Please check `sinon.mock` for this.